Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts

```diff
@@ -0,0 +1,160 @@
+    const diagnostics = get(botDiagnosticsState(projectId));
+    const setting = get(settingsState(projectId));
+    const skillManifests = get(skillManifestsState(projectId));
+    const dialogSchemas = get(dialogSchemasState(projectId));
+    const qnaFiles = get(qnaFilesState(projectId));
+    const formDialogSchemas = get(formDialogSchemasSelectorFamily(projectId));
+    const botProjectFile = get(botProjectFileState(projectId));
+    const jsonSchemaFiles = get(jsonSchemaFilesState(projectId));
+
+    const botAssets: BotAssets = {
+      projectId,
+      dialogs,
+      luFiles,
+      qnaFiles,
+      lgFiles,
+      skillManifests,
+      setting,
+      dialogSchemas,
+      formDialogSchemas,
+      botProjectFile,
+      jsonSchemaFiles,
+    };
+
+    const notifications: Notification[] = [];
+    diagnostics.forEach((d) => {
+      notifications.push(new ServerNotification(projectId, '', d.source, d));
+    });
+    const skillDiagnostics = BotIndexer.checkSkillSetting(botAssets);
+    skillDiagnostics.forEach((item) => {
+      if (item.source.endsWith('.json')) {
+        notifications.push(new SkillNotification(projectId, item.source, item.source, item));
+      } else {
+        notifications.push(new DialogNotification(projectId, item.source, item.source, item));
+      }
+    });
+    const luisLocaleDiagnostics = BotIndexer.checkLUISLocales(botAssets);
+
+    luisLocaleDiagnostics.forEach((item) => {
+      notifications.push(new SettingNotification(projectId, item.source, item.source, item));
+    });
+
+    dialogs.forEach((dialog) => {
+      dialog.diagnostics.forEach((diagnostic) => {
+        const location = `${dialog.id}.dialog`;
+        notifications.push(new DialogNotification(projectId, dialog.id, location, diagnostic));
+      });
+    });
+    getReferredLuFiles(luFiles, dialogs).forEach((lufile) => {
+      lufile.diagnostics.forEach((diagnostic) => {
+        const location = `${lufile.id}.lu`;
+        notifications.push(new LuNotification(projectId, lufile.id, location, diagnostic, lufile, dialogs));
+      });
+    });
+    lgFiles.forEach((lgFile) => {
+      lgFile.diagnostics.forEach((diagnostic) => {
+        const location = `${lgFile.id}.lg`;
+        notifications.push(new LgNotification(projectId, lgFile.id, location, diagnostic, lgFile, dialogs));
+      });
+    });
+    qnaFiles.forEach((qnaFile) => {
+      lodashGet(qnaFile, 'diagnostics', []).forEach((diagnostic) => {
+        const location = `${qnaFile.id}.qna`;
+        notifications.push(new QnANotification(projectId, qnaFile.id, location, diagnostic));
+      });
+    });
+    return notifications;
+  },
+});
+
+export const startAllBotEnableSelector = selector({
+  key: 'startAllBotEnableSelector',
+  get: ({ get }) => {
+    const ids = get(botProjectIdsState);
+    const result = ids.reduce((result: Notification[], id: string) => {
+      return [...result, ...get(diagnosticsSelector(id))];
+    }, []);
+    return !result.length;
```

Comment:
Start Bot button needs to be disabled only if there are errors in root bot or skill. So maybe do a filter on severity before setting it to disabled